### PR TITLE
AG-12204 - Enable no-empty-function eslint rule.

### PR DESCRIPTION
**Pull request by alantreadway**
https://ag-grid.atlassian.net/browse/AG-12204

### CODE REVIEW

File: packages/ag-charts-community/src/api/preset/presetUtils.ts [hunk 1]
```diff
@@ -1,5 +1,12 @@
+import { Logger } from '../../util/logger';
+
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (iMoses) [inline packages/ag-charts-community/src/api/preset/presetUtils.ts]: can't you remove the `eslint-disable` annotation as well?

File: packages/ag-charts-enterprise/src/features/crosshair/crosshair.ts [hunk 2]
```diff
@@ -211,7 +211,7 @@ export class Crosshair extends _ModuleSupport.BaseModuleInstance implements _Mod
             const fractionDigits = (axisLayout?.label.fractionDigits ?? 0) + (isInteger(value) ? 0 : 1);
             return formatNumber(value, fractionDigits);
         }
-        return String(value ?? '');
+        return String(value?.toString() ?? '');
```
Comment (iMoses) [inline packages/ag-charts-enterprise/src/features/crosshair/crosshair.ts]: This doesn't make too much sense 😶 
What was the lint error here?
If a change is needed then I think you can get rid of the outer `String` (i.e. `return value?.toString() ?? '';`

File: packages/ag-charts-community/src/api/preset/presetUtils.ts (hunk 1 above)
Comment (alantreadway) [reply packages/ag-charts-community/src/api/preset/presetUtils.ts]: I thought I'd seen eslint complain about inactive disables, but maybe it was a figment of my imagination :D

Comment (alantreadway) [reply packages/ag-charts-community/src/api/preset/presetUtils.ts]: Enable detection of unused eslint inline disabled, and I've removed everything flagged.

File: packages/ag-charts-enterprise/src/features/crosshair/crosshair.ts (hunk 2 above)
Comment (alantreadway) [reply packages/ag-charts-enterprise/src/features/crosshair/crosshair.ts]: https://typescript-eslint.io/rules/no-base-to-string/

I'm not sure if we should take it tbh, I can undo this change

Comment (alantreadway) [reply packages/ag-charts-enterprise/src/features/crosshair/crosshair.ts]: Reverted the fix

**Pull request merged**